### ISSUE TITLE
MultiIter example should also return pad in next(self)

### ISSUE THREAD

**Issue by yifeim** [apache/incubator-mxnet#9745]
Error caused by eval_data.pad not found:
https://github.com/apache/incubator-mxnet/blob/8205e24d99b95cc2971006f3453a7e7addac7ffe/python/mxnet/module/base_module.py#L295

These templates need to change:
https://mxnet.incubator.apache.org/api/python/io.html#develop-a-new-iterator

I am not quite sure if this example is ``simple'', but it also needs to include pad in return:
https://mxnet.incubator.apache.org/tutorials/basic/data.html#custom-iterator

Thanks,
Yifei

**Comment by karan6181** [comment]
@sandeep-krishnamurthy Can you please add labels:
- Python
- Call for contribution
- Example

@yifeim Can you please create a PR for this with relevant changes so that members can review it.

Thanks!

**Comment by yifeim** [comment]
Hi Karan,

Thanks for attending to this issue. Unfortunately, I ended up using NDArrayIter instead of creating my own. Since I did not find a good use case for a custom DataIter, and additionally, anyone who is really interested in DataIter should write one in C++, e.g., LibSVMIter. I guess I should be working at lower levels if I were to create a PR.

In short, fixing the doc does not solve a more fundamental issue which is whoever decides to write Custom DataIter must have a good reason and reasonable understanding of the codes under the hood. Don't you think so?

Thanks,
Yifei
